cmd/storj-sdk/network.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/zeebo/errs"
+	"golang.org/x/sync/errgroup"
+
+	"storj.io/storj/internal/fpath"
+	"storj.io/storj/internal/processgroup"
+	"storj.io/storj/pkg/utils"
+)
+
+func networkExec(flags *Flags, args []string, command string) error {
+	processes, err := newNetwork(flags.Directory, flags.SatelliteCount, flags.StorageNodeCount)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := NewCLIContext(context.Background())
+	defer cancel()
+
+	err = processes.Exec(ctx, command)
+	closeErr := processes.Close()
+
+	return errs.Combine(err, closeErr)
+}
+
+func networkTest(flags *Flags, command string, args []string) error {
+	processes, err := newNetwork(flags.Directory, flags.SatelliteCount, flags.StorageNodeCount)
+	if err != nil {
+		return err
+	}
+
+	ctx, cancel := NewCLIContext(context.Background())
+
+	var group errgroup.Group
+	processes.Start(ctx, &group, "run")
+
+	time.Sleep(2 * time.Second)
+
+	cmd := exec.CommandContext(ctx, command, args...)
+	cmd.Env = append(os.Environ(), processes.Env()...)
+	stdout := processes.Output.Prefixed("test:out")
+	stderr := processes.Output.Prefixed("test:err")
+	cmd.Stdout, cmd.Stderr = stdout, stderr
+	processgroup.Setup(cmd)
+
+	if printCommands {
+		fmt.Fprintf(processes.Output, "exec: %v\n", strings.Join(cmd.Args, " "))
+	}
+	errRun := cmd.Run()
+
+	cancel()
+	return errs.Combine(errRun, processes.Close(), group.Wait())
+}
+
+func networkDestroy(flags *Flags, args []string) error {
+	if fpath.IsRoot(flags.Directory) {
+		return errors.New("safety check: disallowed to remove root directory " + flags.Directory)
+	}
+	if printCommands {
+		fmt.Println("sdk | exec: rm -rf", flags.Directory)
+	}
+	return os.RemoveAll(flags.Directory)
+}
+
+// newNetwork creates a default network
+func newNetwork(dir string, satelliteCount, storageNodeCount int) (*Processes, error) {
+	processes := NewProcesses()
+
+	const (
+		host            = "127.0.0.1"
+		gatewayPort     = 9000
+		satellitePort   = 10000
+		storageNodePort = 11000
+	)
+
+	defaultSatellite := net.JoinHostPort(host, strconv.Itoa(satellitePort+0))
+
+	arguments := func(name, command, addr string, rest ...string) []string {
+		return append([]string{
+			"--log.level", "debug",
+			"--config-dir", ".",
+			command,
+			"--identity.server.address", addr,
```

Comment:
my last merged pr changed this to --server.address, and then an upcoming pr changes this to --server.public-address, fyi